### ISSUE TITLE
Is it possible to get a cleaner production build output?

### ISSUE THREAD

**Issue by markbenson13** [gatsbyjs/gatsby#16277]
I'm building a gatsby project and what I want is to have a cleaner production build. Do we have ways on how to do it? 

Desired folder structure:

public
  --- js
       -- main.js
  --- css
       -- main.css
  --- index.html

**Comment by wardpeet** [comment]
Sadly, it's not possible. The assets we output are needed to handle all use-cases and optimize your assets.

Why do you care about a cleaner folder structure?

**Comment by markbenson13** [comment]
Mainly for handing it over to back-end devs and then they'll integrate to our CMS which is Concrete5.

**Comment by elken** [comment]
My understanding is you're supposed to use a CMS as a backend data source.

**Comment by m-allanson** [comment]
Hey @markbenson13, @elken's comment is correct - the expected way of using Gatsby would be to use a source plugin to pull data into your Gatsby site.

It looks like there isn't a Concrete5 source plugin, but it does have a REST API, so you would be able to create your own source plugin. Here's some related docs on source plugins:

- [About source plugins](https://www.gatsbyjs.org/tutorial/part-five/#source-plugins)
- [Overview on creating source plugins](https://www.gatsbyjs.org/docs/creating-a-source-plugin/)
- [Source plugin tutorial](https://www.gatsbyjs.org/docs/pixabay-source-plugin-tutorial/)

You _could_ hand over the content of the `public` directory to your backend developers to integrate, but the generated code in the `public` directory will all be minified and optimised. I think this would make it quite difficult to work with. It also means if you make any changes to your Gatsby site, your backend developers will need to manually copy those changes to their integration. I think it'd very quickly become very awkward and time consuming, and make everyone sad 😅.

Using a source plugin means your backend and frontend developers can work independently, and running `gatsby build` will always generate a site with the most up-to-date design and data. 

To summarise - Gatsby has no way to customise the output of the `public` directory, as that directory is intended to be the final 'production artefact' in your project. But Gatsby _is_ designed to work with any backend data source - checkout the docs listed above for more info on that.

I hope that helps!

**Comment by wardpeet** [comment]
Thank you for opening this, @markbenson13 

We're marking this issue as answered and closing it for now but please feel free to reopen this and comment if you would like to continue this discussion. We hope we managed to help and thank you for using Gatsby! 💜

**Comment by markbenson13** [comment]
Thanks for all your answers!
